### ISSUE TITLE
Warning: The uploaded file was only partially uploaded!

### ISSUE THREAD

**Issue by budimanjojo** [bitnami/bitnami-docker-opencart#68]
**Description**

Can not upload theme file. Get the Warning: The uploaded file was only partially uploaded! error. The theme that I'm installing suggest me to change these in php.ini:
```
max_execution_time=3000
max_input_vars = 25000
post_max_size=350M
upload_max_filesize=200M
```
Which I already did by editing the php.ini file in /bitnami/php/conf/php.ini. Restarted the docker container and problem persists.

**Steps to reproduce the issue:**

1. Go to Extensions > Installer > Upload the *.ocmod.zip file (the file is 30MB)
2. Wait for it to upload
3. Error occured

**What actually happens** 

Warning: The uploaded file was only partially uploaded!

**What you expect to happen**

File Successfully uploaded

- Output of `docker version`:

```
Client:
 Version:           18.09.5
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        e8ff056
 Built:             Thu Apr 11 04:43:57 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.5
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       e8ff056
  Built:            Thu Apr 11 04:10:53 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

- Output of `docker info`:

```
Containers: 16
 Running: 13
 Paused: 0
 Stopped: 3
Images: 14
Server Version: 18.09.5
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.15.0
Operating System: Ubuntu 18.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 3GiB
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.24.0, build 0aa59064
docker-py version: 3.7.2
CPython version: 3.6.8
OpenSSL version: OpenSSL 1.1.0j  20 Nov 2018
```

**Comment by tompizmor** [comment]
@budimanjojo can you try specifying those php values in the opencart htaccess file? 

```
/opt/bitnami/apache/conf/vhosts/htaccess/opencart-htaccess.conf
```

As you can see, there are already some placeholders for php settings:

**Comment by budimanjojo** [comment]
Yeah I did. And problem persists. Also, I tried installing smaller
extension from the marketplace and found out there are a lot of permission
errors in the error log (the extension is trying to put files inside
/opt/bitnami/system folder). I dig in a little and found out /opt/bitnami
folder's files are owned by root:root while /bitnami folder's files are
owned by daemon:daemon. Which one should I chown?

**Comment by tompizmor** [comment]
Could you please point me to one of those extensions? I want to try on my side to reproduce the permission issue.

**Comment by budimanjojo** [comment]
Sorry I can't give you the the theme file because it's a paid extension, but here's the link to the product: https://themeforest.net/item/emarket-the-ecommerce-multipurpose-marketplace-opencart-3-theme-mobile-layouts-included/20843842
I already used that theme before moving to docker and it worked. Also, I just tried creating a new container using this image and it installed the theme perfectly (after editing the php.ini file).

**Comment by tompizmor** [comment]
> Also, I just tried creating a new container using this image and it installed the theme perfectly 

Nice. I'm glad it worked at the end.

Regarding the permission issues, I tried with a couple of free extensions from the Opencart marketplace and I was able to reproduce the issue. We will take a look at the permission configuration and update the issue once we have more information.

**Comment by marcosbc** [comment]
Hi @budimanjojo, we have just released [OpenCart `3.0.3-6-debian-10-r65`](https://github.com/bitnami/bitnami-docker-opencart#303-6-debian-10-r65) which includes major changes from previous releases. Now, source code for the initialization logic is published in GitHub in the [`rootfs` folder](https://github.com/bitnami/bitnami-docker-opencart/tree/master/3/debian-10/rootfs) and also introduces significant changes to the initialization approach.

In particular, now the container runs as a non-root user and permissions for PrestaShop files are configured to be writable by the web server. Let us know if this solves your issue!
